encoding/series_iterator.go

```diff
@@ -111,21 +112,33 @@ func (it *seriesIterator) Close() {
 	}
 }
 
-func (it *seriesIterator) Reset(id ident.ID, nsID ident.ID, tags ident.TagIterator, startInclusive, endExclusive time.Time, replicas []Iterator) {
+func (it *seriesIterator) Replicas() []MultiReaderIterator {
+	return it.multiReaderIters
+}
+
+func (it *seriesIterator) Reset(id ident.ID, nsID ident.ID, tags ident.TagIterator, startInclusive, endExclusive time.Time, replicas []MultiReaderIterator) {
 	it.id = id
 	it.nsID = nsID
 	it.tags = tags
 	it.start = startInclusive
 	it.end = endExclusive
 	it.iters.reset()
 	it.iters.setFilter(startInclusive, endExclusive)
+
+	for idx := range it.multiReaderIters {
```

Comment:
So pooling `[]MultiReaderIterator` introduces a little more nuance to this:
- `seriesIterator.Reset()` must transfer the ownership of the `[]MultiReaderIterator` from the caller to `seriesIterator`
- This means the calls to `.Put` the slice back into the pool that are currently done in `client/session.go` must only be done in case of error; in case of success, `.Put` must be done by `seriesIterator.Close()`
- This means the seriesIterator needs to know about the pool (i.e. `NewSeriesIterator` needs to take a param for the `MultiReaderIteratorArrayPool`

Welcome to the joys of pooling without malloc/destructors :D

Reply:
Actually, scratch all that. I think the only change that's required is:
- moving the `it.multiReaderIters` cleanup (for loop setting to nil/setting to slice of len 0) to Close()
- ensure you do `it.multiReaderIters = it.multiReaderIters[:0]` in `Reset()`

This way pooling the `seriesIterator` pools the underlying slices too.